tests/test_date.py

```diff
@@ -239,42 +413,37 @@ def test_should_return_yesterday(self, date_string):
         param('Avant-hier'),
     ])
     def test_should_return_day_before_yesterday(self, date_string):
-        today = datetime.utcnow().replace(hour=0, minute=0, second=0, microsecond=0)
-        day_before_yesterday = today - relativedelta(days=2)
-        date_data = self.parser.get_date_data(date_string)
-        self.assertIsNotNone(date_data['date_obj'])
-        self.check_equal(day_before_yesterday.date(),
-                         date_data['date_obj'].date(), date_string)
+        self.given_date_string(date_string)
+        self.given_today()
+        self.given_day_before_yesterday()
+        self.when_date_data_is_parsed()
+        self.then_check_date()
+        self.then_day_before_yesterday_was_parsed()
 
     def test_should_not_assume_language_too_early(self):
-        date_fixtures = [
-            (u'07/07/2014', datetime(2014, 7, 7)),
-            (u'07.jul.2014 | 12:52', datetime(2014, 7, 7)),
-            (u'07.ago.2014 | 12:52', datetime(2014, 8, 7)),
-            (u'07.feb.2014 | 12:52', datetime(2014, 2, 7)),
-            (u'07.ene.2014 | 12:52', datetime(2014, 1, 7)),
-        ]
+        dates_to_parse = {u'07/07/2014': datetime(2014, 7, 7),
+                          u'07.jul.2014 | 12:52': datetime(2014, 7, 7),
+                          u'07.ago.2014 | 12:52': datetime(2014, 8, 7),
+                          u'07.feb.2014 | 12:52': datetime(2014, 2, 7),
+                          u'07.ene.2014 | 12:52': datetime(2014, 1, 7)}
 
-        for date_string, correct_date in date_fixtures:
-            date_data = self.parser.get_date_data(date_string)
-            self.assertIsNotNone(date_data['date_obj'],
-                                 "Could not get date for: %s" % date_string)
-            self.check_equal(correct_date.date(),
-                             date_data['date_obj'].date(), date_string)
+        self.given_multiple_dates(dates_to_parse)
+        self.when_multiple_dates_are_parsed()
+        self.then_check_language_too_early()
+        self.then_language_too_early_was_parsed()
 
     def test_should_enable_redetection_for_multiple_languages(self):
-        date_fixtures = [
-            (u'13 Ago, 2014', datetime(2014, 8, 13)),
-            (u'11 Marzo, 2014', datetime(2014, 3, 11)),
-            (u'13 Septiembre, 2014', datetime(2014, 9, 13)),
-            (u'13 Setembro, 2014', datetime(2014, 9, 13)),
-            (u'13 Março, 2014', datetime(2014, 3, 13)),
-        ]
-        parser = date.DateDataParser(allow_redetect_language=True)
-
-        for date_string, correct_date in date_fixtures:
-            date_data = parser.get_date_data(date_string)
-            self.assertEqual(correct_date.date(), date_data['date_obj'].date())
+        dates_to_parse = {u'11 Marzo, 2014': datetime(2014, 3, 11),
```

Comment:
You've ordered them correctly in the code, yes. But do you remember how we discussed that mappings do not preserve the order?